cmd/gateway-router.go

```diff
@@ -67,63 +67,63 @@ func registerGatewayAPIRouter(mux *router.Router, gw GatewayLayer) {
 	// DeleteObject
-	bucket.Methods("DELETE").Path("/{object:.+}").HandlerFunc(api.DeleteObjectHandler)
+	bucket.Methods("DELETE").Path("/{object:.+}").HandlerFunc(httpTracelogAll(api.DeleteObjectHandler))
 
 	/// Bucket operations
 
 	// GetBucketLocation
-	bucket.Methods("GET").HandlerFunc(api.GetBucketLocationHandler).Queries("location", "")
+	bucket.Methods("GET").HandlerFunc(httpTracelogAll(api.GetBucketLocationHandler)).Queries("location", "")
 	// GetBucketPolicy
-	bucket.Methods("GET").HandlerFunc(api.GetBucketPolicyHandler).Queries("policy", "")
+	bucket.Methods("GET").HandlerFunc(httpTracelogAll(api.GetBucketPolicyHandler)).Queries("policy", "")
 	// GetBucketNotification
-	bucket.Methods("GET").HandlerFunc(api.GetBucketNotificationHandler).Queries("notification", "")
+	bucket.Methods("GET").HandlerFunc(httpTracelogAll(api.GetBucketNotificationHandler)).Queries("notification", "")
 	// ListenBucketNotification
-	bucket.Methods("GET").HandlerFunc(api.ListenBucketNotificationHandler).Queries("events", "{events:.*}")
+	bucket.Methods("GET").HandlerFunc(httpTracelogAll(api.ListenBucketNotificationHandler)).Queries("events", "{events:.*}")
 	// ListMultipartUploads
-	bucket.Methods("GET").HandlerFunc(api.ListMultipartUploadsHandler).Queries("uploads", "")
+	bucket.Methods("GET").HandlerFunc(httpTracelogAll(api.ListMultipartUploadsHandler)).Queries("uploads", "")
 	// ListObjectsV2
-	bucket.Methods("GET").HandlerFunc(api.ListObjectsV2Handler).Queries("list-type", "2")
+	bucket.Methods("GET").HandlerFunc(httpTracelogAll(api.ListObjectsV2Handler)).Queries("list-type", "2")
 	// ListObjectsV1 (Legacy)
-	bucket.Methods("GET").HandlerFunc(api.ListObjectsV1Handler)
+	bucket.Methods("GET").HandlerFunc(httpTracelogAll(api.ListObjectsV1Handler))
 	// PutBucketPolicy
-	bucket.Methods("PUT").HandlerFunc(api.PutBucketPolicyHandler).Queries("policy", "")
+	bucket.Methods("PUT").HandlerFunc(httpTracelogAll(api.PutBucketPolicyHandler)).Queries("policy", "")
 	// PutBucketNotification
-	bucket.Methods("PUT").HandlerFunc(api.PutBucketNotificationHandler).Queries("notification", "")
+	bucket.Methods("PUT").HandlerFunc(httpTracelogAll(api.PutBucketNotificationHandler)).Queries("notification", "")
 	// PutBucket
-	bucket.Methods("PUT").HandlerFunc(api.PutBucketHandler)
+	bucket.Methods("PUT").HandlerFunc(httpTracelogAll(api.PutBucketHandler))
 	// HeadBucket
-	bucket.Methods("HEAD").HandlerFunc(api.HeadBucketHandler)
+	bucket.Methods("HEAD").HandlerFunc(httpTracelogAll(api.HeadBucketHandler))
 	// PostPolicy
-	bucket.Methods("POST").HeadersRegexp("Content-Type", "multipart/form-data*").HandlerFunc(api.PostPolicyBucketHandler)
+	bucket.Methods("POST").HeadersRegexp("Content-Type", "multipart/form-data*").HandlerFunc(httpTracelogAll(api.PostPolicyBucketHandler))
 	// DeleteMultipleObjects
-	bucket.Methods("POST").HandlerFunc(api.DeleteMultipleObjectsHandler)
+	bucket.Methods("POST").HandlerFunc(httpTracelogAll(api.DeleteMultipleObjectsHandler))
```

Comment:
Once we move `httpTraceLog` to its own package we can simplify these and their naming conventions.